### DISCUSSION TITLE
Featured Image Not Displaying Properly on Single Post

### DISCUSSION THREAD

**Discussion by milesdame** [wprig/wprig#801] [Q&A] [answered]
Hello, 

I am having an issue that might be related to wp-rig. Right now my featured image works correctly on my home page (which is just a list of posts). On the home page the featured image is cropped to be a square (how I want it), but for some reason when I view a single post it remains cropped as a square when I really want it to be full sized. From everything I've read it seems like I'm doing this correctly, so I thought maybe I'm not fully understanding our wp-rig interprets things. Any help would be appreciated. Thanks :) 

I've tried making the following changes:

To component.php I added an image size: `add_image_size( 'wp-rig-single', 1110, 555, true ); // Single Post featured image`

Then in entry_thumbnail.php I've tried setting up the following code: 

```
if ( is_single() ) {
	?>
	<div class="post-thumbnail">
		<?php
		if ( has_post_thumbnail() ) {
			the_post_thumbnail(
				'wp-rig-single',
				array(
					'alt' => the_title_attribute(
						array(
							'echo' => false,
						)
					),
				)
			);
		} else {
			echo '<img loading="lazy" src="' . esc_attr( get_template_directory_uri() ) . '/assets/images/single-post-fall-back.jpg" class="size-post-thumbnail attachment-post-thumbnail" aria-hidden="true">';
		}
		?>
	</div><!-- .post-thumbnail -->
	<?php
} else {
	?>
	<div class="post-thumbnail">
		<?php
		if ( has_post_thumbnail() ) {
			the_post_thumbnail(
				'wp-rig-featured',
				array(
					'alt' => the_title_attribute(
						array(
							'echo' => false,
						)
					),
				)
			);
		} else {
			echo '<img loading="lazy" src="' . esc_attr( get_template_directory_uri() ) . '/assets/images/fall-back.jpg" class="size-post-thumbnail attachment-post-thumbnail" aria-hidden="true">';
		}
		?>
	</div><!-- .post-thumbnail -->
	<?php
}
```

**Comment by dthenley** [comment, marked as answer]
Hey @milesdame  where did you add the custom image size? I would suggest adding it in the `Post_Thumbnails\Component.php` file.https://github.com/wprig/wprig/blob/34d005a6deaa2cbf65f86e4a18d73832aeda2b02/inc/Post_Thumbnails/Component.php#L50

Also make sure that you regenerate the thumbnail after you set up the new image size.

**Reply by milesdame** [reply]
Hey @dthenley thanks for the reply. I added the custom image size to `Post_Thumbnails\Component.php` and I regenerated the thumbnails, but it has no effect on them at all :(

This is my `Post_Thumbnails\Component.php`:
```
/**
	 * Adds custom image sizes.
	 */
	public function action_add_image_sizes() {
		add_image_size( 'wp-rig-featured', 480, 480, true );
		add_image_size( 'wp-rig-single', 1110, 555, false );
	}
```

**Reply by dthenley** [reply]
Welp it look like you did all that part right. My only other thing I could think of is if the image is big enough to fit into your dimensions. If the image size or height is doesn't meet the width or height of the custom image size then it won't work as well.

Also you can check to see if it is being called correctly by checking the class of the image to see if it has a class of `size-wp-rig-single`

**Reply by milesdame** [reply]
I feel so incredibly stupid... I just realized what the issue is. I was editing the wp-rig code, but I didn't have that theme activated in my test environment....

I do appreciate you taking time to help me trouble shoot though! :)

**Reply by dthenley** [reply]
LMAO! I have definitely been there! Well I'm glad you got it working!

**Comment by robruiz** [comment]
I would also consider the possibility of CSS forcing the size in some way.

On Mon, May 2, 2022 at 9:45 AM Donte ***@***.***> wrote:

> Welp it look like you did all that part right. My only other thing I could
> think of is if the image is big enough to fit into your dimensions. If the
> image size or height is doesn't meet the width or height of the custom
> image size then it won't work as well.
>
> Also you can check to see if it is being called correctly by checking the
> class of the image to see if it has a class of size-wp-rig-single
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/wprig/wprig/discussions/801#discussioncomment-2674205>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AB6LMGPRZLK7ZHPB6XKYSRTVH7S7PANCNFSM5UWZMR3Q>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Rob Ruiz
Owner of Bizzle Designs
Web Design/Graphic Design/Sound Design/Audio & Video Production
www.bizzledesigns.com
